substrate/primitives/runtime/src/generic/unchecked_extrinsic.rs

```diff
@@ -122,66 +297,137 @@ where
+
+	/// New instance of an old-school signed transaction.
+	pub fn new_signed(
+		function: Call,
+		signed: Address,
+		signature: Signature,
+		tx_ext: Extension,
+	) -> Self {
+		Self {
+			preamble: Preamble::Signed(
+				signed,
+				signature,
+				EXTENSION_VERSION,
+				tx_ext,
+				EXTRINSIC_FORMAT_VERSION,
+			),
+			function,
+		}
+	}
+
+	/// New instance of an old-school signed transaction on extrinsic format version 4.
+	pub fn new_signed_legacy(
+		function: Call,
+		signed: Address,
+		signature: Signature,
+		tx_ext: Extension,
+	) -> Self {
+		Self {
+			preamble: Preamble::Signed(
+				signed,
+				signature,
+				0,
+				tx_ext,
+				LOWEST_SUPPORTED_EXTRINSIC_FORMAT_VERSION,
+			),
+			function,
+		}
+	}
+
+	/// New instance of an new-school unsigned transaction.
+	pub fn new_transaction(function: Call, tx_ext: Extension) -> Self {
+		Self { preamble: Preamble::General(EXTENSION_VERSION, tx_ext), function }
 	}
 }
 
-impl<LookupSource, AccountId, Call, Signature, Extra, Lookup> Checkable<Lookup>
-	for UncheckedExtrinsic<LookupSource, Call, Signature, Extra>
+impl<Address: TypeInfo, Call: TypeInfo, Signature: TypeInfo, Extension: TypeInfo> ExtrinsicLike
+	for UncheckedExtrinsic<Address, Call, Signature, Extension>
+{
+	fn is_bare(&self) -> bool {
+		matches!(self.preamble, Preamble::Bare(_))
+	}
+
+	fn is_signed(&self) -> Option<bool> {
+		Some(matches!(self.preamble, Preamble::Signed(..)))
+	}
+}
+
+// TODO: Migrate existing extension pipelines to support current `Signed` transactions as `General`
+// transactions by adding an extension to validate signatures, as they are currently validated in
+// the `Checkable` implementation for `Signed` transactions.
+
+impl<LookupSource, AccountId, Call, Signature, Extension, Lookup> Checkable<Lookup>
+	for UncheckedExtrinsic<LookupSource, Call, Signature, Extension>
 where
 	LookupSource: Member + MaybeDisplay,
-	Call: Encode + Member,
+	Call: Encode + Member + Dispatchable,
 	Signature: Member + traits::Verify,
 	<Signature as traits::Verify>::Signer: IdentifyAccount<AccountId = AccountId>,
-	Extra: SignedExtension<AccountId = AccountId>,
+	Extension: Encode + TransactionExtension<Call>,
 	AccountId: Member + MaybeDisplay,
 	Lookup: traits::Lookup<Source = LookupSource, Target = AccountId>,
 {
-	type Checked = CheckedExtrinsic<AccountId, Call, Extra>;
+	type Checked = CheckedExtrinsic<AccountId, Call, Extension>;
 
 	fn check(self, lookup: &Lookup) -> Result<Self::Checked, TransactionValidityError> {
-		Ok(match self.signature {
-			Some((signed, signature, extra)) => {
+		Ok(match self.preamble {
+			Preamble::Signed(signed, signature, _, tx_ext, tx_version) => {
 				let signed = lookup.lookup(signed)?;
-				let raw_payload = SignedPayload::new(self.function, extra)?;
+				// The `Implicit` is "implicitly" included in the payload.
+				let raw_payload = match tx_version {
+					0..=LOWEST_SUPPORTED_EXTRINSIC_FORMAT_VERSION =>
+						SignedPayload::new_legacy(self.function, tx_ext)?,
```

Comment:
```suggestion
					LOWEST_SUPPORTED_EXTRINSIC_FORMAT_VERSION =>
						SignedPayload::new_legacy(self.function, tx_ext)?,
```

You can not just accept `0..4` as valid tx versions.

Reply:
Fixed.